### PR TITLE
Middleware runs according to path not router

### PR DESCRIPTION
**Pull request by tunniclm**
This pr attempts to document this issue: https://github.com/expressjs/express/issues/2760

Feedback very welcome!

cc: @crandmck

### CODE REVIEW

File: _includes/api/en/4x/router-use.md [hunk 1]
```diff
@@ -77,3 +77,25 @@ app.use(express.static(__dirname + '/uploads'));
...
+
+authRouter.use(require('./authenticate').basic(usersdb));
+
+authRouter.get('/:user_id/edit', function(req, res, next) { ... edit user ui ... });
+openRouter.get('/', function(req, res, next) { ... list users ... })
+openRouter.get('/:user_id', function(req, res, next) { ... view user ... })
+
+app.use('/users', authRouter);
+app.use('/users', openRouter);
+```
```
Comment (tunniclm) [inline _includes/api/en/4x/router-use.md]: Ah, forgot to convert this code block to the right format.

**Comment by crandmck** [comment]
Thanks for the clarification.  It's usually easier for me if you can branch and then open a PR from the branch so I can make edits there and then merge the final form.  For now, I'll merge and then edit in `gh-pages`... unless there is better workflow that I don't know about (entirely possible since I'm no Git expert).

**Comment by tunniclm** [comment]
@crandmck There was a branch on my repo (tunniclm:note_on_router_middleware) and I opened the PR from that branch. Was there something else I was missing?

**Comment by crandmck** [comment]
I think `tunniclm:note_on_router_middleware` is a fork, not a branch.  I just need to be able to edit the files on that branch, before merging the PR, which I can't do with that.  I guess you need to have collaborator rights to create a new branch....?  Perhaps I could clone that branch on `https://github.com/tunniclm/expressjs.com`. I think I just need to understand the required workflow.

**Comment by hacksparrow** [comment]
@tunniclm how @crandmck and I have been working is, we create a branch on **this repo**, and collaborate on edits till the content on the branch is ready to be merged. His technical writing experience helps to give the professional touch we need in the docs.

Sian and Mike will need write access to be able to follow the workflow we've been following.

**Comment by crandmck** [comment]
Is there any reaoson why @tunniclm and @sjanuary shouldn't have commit access to this repo?

**Pull request merged**